File: Data Structures & Algorithms/is-subsequence/submission-0.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        i = j = 0
        n1 = len(s)
        n2 = len(t)
        while i<n1 and j<n2:
            if s[i]==t[j]:
                i=i+1
            j=j+1
        return i==n1
        

File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-0.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string,int>mp;
        for(auto s: arr){
            mp[s]++;
        }
        int cnt = 0;
        for(auto a: arr){
            if(mp[a]>1){
                continue;
            }
            else{
                cnt++;
                if(cnt==k){
                    return a;
                }
            }
        }
        return "";
    }
};